tests/integration/tests/test_util/util.py

```diff
@@ -339,6 +339,11 @@ def is_node_ready(
         LOG.info(f"Node not ready yet: {node_name}, failed to retrieve node info: {ex}")
         return False
 
+    # Expect all snap services to be up and running. Add a little sleep to
+    # ensure that services not just started but are also stable.
+    time.sleep(5)
+    check_snap_services(node_name)
```

Comment:
I'd catch exceptions raised by ``check_snap_services`` and just return False since that's what the callers expect.

You could move it in the try/except block above and update the exception message to ``LOG.info(f"Node not ready yet: {node_name}, exception: {ex}")